[autoscaler] Implement node provider for aliyun (#15712)
ray-project/ray: doc/source/cluster/cloud.rst

    .. group-tab:: Aliyun

        First, you need to configure your aliyun account. You need to set access key and access key secret in yaml config file.

        Once above done, you should be ready to launch your cluster. The provided `ray/python/ray/autoscaler/aliyun/example-full.yaml </ray/python/ray/autoscaler/aliyun/example-full.yaml>`__ cluster config file will create a small cluster with an ecs.n4.large head node (on-demand) configured to autoscale up to two ecs.n4.2xlarge.

        Test that it works by running the following commands from your local machine:

        .. code-block:: bash

            # Create or update the cluster. When the command finishes, it will print
            # out the command that can be used to SSH into the cluster head node.
            $ ray up ray/python/ray/autoscaler/aliyun/example-full.yaml

            # Get a remote screen on the head node.
            $ ray attach ray/python/ray/autoscaler/aliyun/example-full.yaml
            $ # Try running a Ray program with 'ray.init(address="auto")'.

            # Tear down the cluster.
            $ ray down ray/python/ray/autoscaler/aliyun/example-full.yaml

> 

Reviewer: @zhuangzhuang131419 
Thanks for the contribution! I think it would be great if the new documentation directed users to you and other maintainers for issues specific to this node provider.

Author: ok. Sorry, I am busy in a training recently, my colleague chenk008 will help me fix it.

Reviewer 2: I have add maintainer info in that section

            # Tear down the cluster
            $ ray down ray/python/ray/autoscaler/staroid/example-full.yaml

    .. group-tab:: Aliyun

>         First, you need to configure your aliyun account. You need to set access key and access key secret in yaml config file.

Reviewer 3: Can you add some notes as to how to do this?

Reviewer 2: I have added description about how to set up AccessKey pair.

>         Once above done, you should be ready to launch your cluster. The provided `ray/python/ray/autoscaler/aliyun/example-full.yaml </ray/python/ray/autoscaler/aliyun/example-full.yaml>`__ cluster config file will create a small cluster with an ecs.n4.large head node (on-demand) configured to autoscale up to two ecs.n4.2xlarge.

Reviewer 3: ```suggestion
        Once the above is done, you should be ready to launch your cluster. The provided `ray/python/ray/autoscaler/aliyun/example-full.yaml </ray/python/ray/autoscaler/aliyun/example-full.yaml>`__ cluster config file will create a small cluster with an `ecs.n4.large` head node (on-demand) configured to autoscale up to two `ecs.n4.2xlarge` nodes.
```

>         Make sure account balance is not less than 100 RMB, otherwise you will got InvalidAccountStatus.NotEnoughBalance error.

Reviewer 3: ```suggestion
        Make sure your account balance is not less than 100 RMB, otherwise you will receive a `InvalidAccountStatus.NotEnoughBalance` error.
```

> 
>         First, install the aliyun client package (``pip install aliyun-python-sdk-core aliyun-python-sdk-ecs``), then obtain the AccessKey pair of the Aliyun account as described in `the docs <https://www.alibabacloud.com/help/en/doc-detail/175967.htm>` and garnt AliyunECSFullAccess/AliyunVPCFullAccess permissions to the RAM user, set the AccessKey pair in yaml config file.
> 
>         Once the above is done, you should be ready to launch your cluster. The provided `ray/python/ray/autoscaler/aliyun/example-full.yaml </ray/python/ray/autoscaler/aliyun/example-full.yaml>`__ cluster config file will create a small cluster with an `ecs.n4.large` head node (on-demand) configured to autoscale up to two `ecs.n4.2xlarge` nodes.

Reviewer 3: ```suggestion

        First, install the aliyun client package (``pip install aliyun-python-sdk-core aliyun-python-sdk-ecs``). Obtain the AccessKey pair of the Aliyun account as described in `the docs <https://www.alibabacloud.com/help/en/doc-detail/175967.htm>`__ and grant AliyunECSFullAccess/AliyunVPCFullAccess permissions to the RAM user. Finally, set the AccessKey pair in your cluster config file.

        Once the above is done, you should be ready to launch your cluster. The provided `aliyun/example-full.yaml </ray/python/ray/autoscaler/aliyun/example-full.yaml>`__ cluster config file will create a small cluster with an ``ecs.n4.large`` head node (on-demand) configured to autoscale up to two ``ecs.n4.2xlarge`` nodes.
```
